### ISSUE TITLE
Issue with Windows using aws cli with python 3 installer

### ISSUE THREAD

**Issue by repl-mike-roest** [oktadeveloper/okta-aws-cli-assume-role#151]
AWSCLI has a new installer for windows that installs a python3 version of the cli.  It is no longer a exe, and is instead a .cmd batch script.  This seems to cause issues with the okta tool.

Previous version that works:
C:\Users\mike>aws --version
aws-cli/1.11.129 Python/2.7.9 Windows/8 botocore/1.5.92

New version that does not
C:\Users\mike>aws --version
aws-cli/1.15.66 Python/3.6.0 Windows/10 botocore/1.10.65

I can login with user password and MFA.  It then will give my profile selection and gives the following error:

Exception in thread "main" java.io.IOException: Cannot run program "aws": CreateProcess error=2, The system cannot find the file specified
        at java.base/java.lang.ProcessBuilder.start(Unknown Source)
        at java.base/java.lang.ProcessBuilder.start(Unknown Source)
        at com.okta.tools.awscli.main(awscli.java:39)
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
        at java.base/java.lang.ProcessImpl.create(Native Method)
        at java.base/java.lang.ProcessImpl.<init>(Unknown Source)
        at java.base/java.lang.ProcessImpl.start(Unknown Source)
        ... 3 more

Looking at the AWSCLI, it's now a .cmd file with the following contents:

```
@rem
@echo off

setlocal

"%~dp0/../runtime/python.exe" -m awscli %*
```

My path contains the awscli location and I can run it directly from a command prompt as my PATHEXT has .cmd
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.RB;.RBW

The python 3 installer can be gotten from https://docs.aws.amazon.com/cli/latest/userguide/awscli-install-windows.html looking for Upgraded MSI installers Are Available header.

**Comment by AlainODea** [comment]
@repl-mike-roest the best workaround I can think is to use PowerShell and the following in your PowerShell profile:

```powershell
function With-Okta {
    Param([string]$Profile)
    Write-Host $args
    $OriginalOKTA_PROFILE = $env:OKTA_PROFILE
    try {
        $env:OKTA_PROFILE = $Profile
        $InternetOptions = Get-ItemProperty -Path "HKCU:\Software\Microsoft\Windows\CurrentVersion\Internet Settings"
        ($ProxyHost, $ProxyPort) = $InternetOptions.ProxyServer.Split(":")
        if ($InternetOptions.ProxyOverride) {
            $NonProxyHosts = [System.String]::Join("|", ($InternetOptions.ProxyOverride.Replace("<local>", "").Split(";") | Where-Object {$_}))
        } else {
            $NonProxyHosts = ""
        }
        java "-Dhttp.proxyHost=$ProxyHost" "-Dhttp.proxyPort=$ProxyPort" "-Dhttps.proxyHost=$ProxyHost" "-Dhttps.proxyPort=$ProxyPort" "-Dhttp.nonProxyHosts=$NonProxyHosts" -classpath $HOME\.okta\* com.okta.tools.WithOkta @args
    } finally {
        $env:OKTA_PROFILE = $OriginalOKTA_PROFILE
    }
}
function aws {
    Param([string]$Profile)
    With-Okta -Profile $Profile aws.cmd --profile $Profile @args
}
```

Once you have that in place you should be able to run commands like this:

```powershell
PS > aws test sts get-caller-identity
```

It should run this after establishing or reusing your existing Okta and AWS session named **test**:

```powershell
aws.cmd --profile test sts get-caller-identity
```

Give that a shot and let me know if it works for you.

**Comment by AlainODea** [comment]
@repl-mike-roest how are you installing AWS CLI? I just spun up a new Windows VM, installed using 64-bit Windows Installer from https://aws.amazon.com/cli/ and it worked.

I'll try it directly with pip now. Is there a particular reason why the Windows installer won't work for your environment?

**Comment by AlainODea** [comment]
I installed Python 3.7 x86_64 for Windows using the web-based installer and chose to add Python 3.7 to PATH (I figure that's what people here are doing since otherwise not even pip is on the path).

This works (I updated it to fix a proxy selection bug #153):
https://github.com/oktadeveloper/okta-aws-cli-assume-role/issues/151#issuecomment-408704347

**Comment by repl-mike-roest** [comment]
Thanks @AlainODea
